flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecGroupWindowAggregate.java

```diff
@@ -0,0 +1,373 @@
+import org.apache.flink.table.runtime.operators.window.WindowOperatorBuilder;
+import org.apache.flink.table.runtime.types.LogicalTypeDataTypeConverter;
+import org.apache.flink.table.runtime.typeutils.InternalTypeInfo;
+import org.apache.flink.table.types.DataType;
+import org.apache.flink.table.types.logical.LogicalType;
+import org.apache.flink.table.types.logical.RowType;
+
+import org.apache.calcite.rel.core.AggregateCall;
+import org.apache.calcite.tools.RelBuilder;
+import org.apache.commons.lang3.ArrayUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.apache.flink.table.planner.plan.utils.AggregateUtil.hasRowIntervalType;
+import static org.apache.flink.table.planner.plan.utils.AggregateUtil.hasTimeIntervalType;
+import static org.apache.flink.table.planner.plan.utils.AggregateUtil.isProctimeAttribute;
+import static org.apache.flink.table.planner.plan.utils.AggregateUtil.isRowtimeAttribute;
+import static org.apache.flink.table.planner.plan.utils.AggregateUtil.isTableAggregate;
+import static org.apache.flink.table.planner.plan.utils.AggregateUtil.toDuration;
+import static org.apache.flink.table.planner.plan.utils.AggregateUtil.toLong;
+import static org.apache.flink.table.planner.plan.utils.AggregateUtil.transformToStreamAggregateInfoList;
+
+/** Stream {@link ExecNode} for either group window aggregate or group window table aggregate. */
+public class StreamExecGroupWindowAggregate extends ExecNodeBase<RowData>
+        implements StreamExecNode<RowData> {
+    private static final Logger LOGGER =
+            LoggerFactory.getLogger(StreamExecGroupWindowAggregate.class);
+
+    private final int[] grouping;
+    private final AggregateCall[] aggCalls;
+    private final LogicalWindow window;
+    private final PlannerNamedWindowProperty[] namedProperties;
+    private final int inputTimeFieldIndex;
+    private final WindowEmitStrategy emitStrategy;
+    private final boolean needRetraction;
+    private final String aggType;
+
+    public StreamExecGroupWindowAggregate(
+            int[] grouping,
+            AggregateCall[] aggCalls,
+            LogicalWindow window,
+            PlannerNamedWindowProperty[] namedProperties,
+            int inputTimeFieldIndex,
+            WindowEmitStrategy emitStrategy,
+            boolean needRetraction,
+            String aggType,
+            ExecEdge inputEdge,
+            RowType outputType,
+            String description) {
+        super(Collections.singletonList(inputEdge), outputType, description);
+        this.grouping = grouping;
+        this.aggCalls = aggCalls;
+        this.window = window;
+        this.namedProperties = namedProperties;
+        this.inputTimeFieldIndex = inputTimeFieldIndex;
+        this.emitStrategy = emitStrategy;
+        this.needRetraction = needRetraction;
+        this.aggType = aggType;
```

Comment:
I think `aggType` is a too generic name. It seems only be used for printing. Is it possible to eliminate it? Maybe we don't need to mention this is a TableAggregate or Aggregate, the important information is that this is a window.